dali/pipeline/operators/fused/crop_cast_permute.cc

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2017-2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/pipeline/operators/fused/crop_cast_permute.h"
+
+namespace dali {
+
+DALI_SCHEMA(CropCastPermute)
+  .DocStr(R"code(Perform a random crop, data type
+cast and permute (from NHWC to NCHW).)code")
+  .NumInput(1)
+  .NumOutput(1)
+  .AllowMultipleInputSets()
+  .AddOptionalArg("output_dtype",
+      R"code(Output data type.)code", DALI_NO_TYPE)
```

Comment:
If DALI_NO_TYPE is a default value of this parameter then it may as well be a required argument. Make it consistent with CropMirrorNormalize.